Added input validation to metadata sent using REST endpoints

cmagnobarbosa opened the pull request:
### :octocat: Are you working on some issue? Identify the issue!

Fix #144 


### :bookmark_tabs: Description of the Change

Add input validation to metadata sent in the payload at REST endpoints,
 this modification avoids broken related to wrong format.

### :computer: Verification Process

- Unit tests performed and verified manually.

### :page_facing_up: Release Notes

- Add input validation to metadata

cmagnobarbosa committed:
Add input validation to metadata sent using REST endpoints

Add input validation to metadata sent in the payload at REST
 endpoints, this modification avoids broken related to wrong format.

cmagnobarbosa committed:
Added unit tests to test the metadata format

hdiogenes commented on main.py:
```diff
@@ -280,7 +281,14 @@ def get_switch_metadata(self, dpid):
+            return jsonify('The request body is not a well-formed JSON.'), 400
+        if metadata is None:
+            return jsonify('The request body mimetype is not'
+                           ' application/json.'), 400
```
If the metadata is empty, the error will always be the mime type? Is this right?

cmagnobarbosa replied:
I added a check to the empty request case(https://github.com/kytos/topology/pull/148/commits/0271d10862ba8934a35442e5427074298ead1b02), more information related to the content type error was also added.

hdiogenes commented on main.py:
```diff
@@ -474,7 +488,14 @@ def get_link_metadata(self, link_id):
+            return jsonify('The request body is not a well-formed JSON.'), 400
+        if metadata is None:
+            return jsonify('The request body mimetype is not'
+                           ' application/json.'), 400
```
Does this really need to be repeated 3 times? Can it be refactored into a function?

cmagnobarbosa replied:
This check has been refactored into a function.

cmagnobarbosa committed:
Add method to get and verify metadata format

Add a method to get metadata sent
 by REST endpoint avoids code duplication and
check the metadata format.

cmagnobarbosa committed:
Added empty metadata error handler

Added empty error handler and the error message was improved.

hdiogenes commented on main.py:
```diff
@@ -55,6 +56,24 @@ def shutdown(self):
+            result = 'The request body is not a well-formed JSON.'
+            raise BadRequest(result)
+        if content_type is None:
+            result = ('The request body is empty.')
```
```suggestion
            result = 'The request body is empty.'
```

hdiogenes commented on main.py:
```diff
@@ -55,6 +56,24 @@ def shutdown(self):
+            result = ('The request body is empty.')
+            raise BadRequest(result)
+        if metadata is None:
+            result = ('The content type must be application/json and not '
+                      f'{content_type}.')
```
Same problem as before: you're checking for 'empty metadata' e telling the user the problem is the 'content-type'. 

```suggestion
            if content_type != 'application/json':
                result = ('The content type must be application/json '
                          f'(received {content_type}).')
            else:
                result = 'Metadata is empty.'
```

cmagnobarbosa committed:
Apply suggestions from code review

Co-authored-by: Humberto Diógenes <hdiogenes@gmail.com>
